pkg/crypto/rings/errors.go

```diff
@@ -0,0 +1,10 @@
+package rings
+
+import (
+	sdkerrors "cosmossdk.io/errors"
+)
+
+var (
+	codespace          = "rings"
+	ErrRingsWrongCurve = sdkerrors.Register(codespace, 1, "key is not a secp256k1 public key")
```

Comment:
s/ErrRingsWrongCurve/ErrRingsNotSecp256

The message is very specific but the type could be misused.